### PR TITLE
Feat(16): Implement blog filter

### PR DESCRIPTION
**Pull request by naseerkassar**
Please checkout the new PR(#16). Bear in mind that my prettier doesn't work for the jsx files (you saw it yesterday) incase there is prettier issues. I will appreciate the feedback incase any is needed
thank in advance :)

### CODE REVIEW

File: src/posts/model.js [hunk 1]
```diff
@@ -17,6 +17,9 @@ export const PostBaseData = graphql`
     intro
     language
     slug
+    category {
+      name
+    }
```
Comment (Moeriki) [inline src/posts/model.js]: Alphabetic.

File: src/posts/model.js [hunk 2]
```diff
@@ -31,6 +34,7 @@ export const postShape = {
   intro: string,
   language: string,
   slug: string,
+  category: shape({ name: string }),
```
Comment (Moeriki) [inline src/posts/model.js]: Keys should be alphabetic.

File: src/posts/model.js [hunk 3]
```diff
@@ -28,13 +28,13 @@ export const PostBaseData = graphql`
 `;
 
 export const postShape = {
-  id: string,
+  category: shape({ name: string }),
   date: string,
   fields: shape({ remote_hero_image: shape({}) }),
+  id: string,
```
Comment (Moeriki) [inline src/posts/model.js]: Keep id at the top.

File: src/posts/model.js (hunk 1 above)
Comment (naseerkassar) [reply src/posts/model.js]: Fixed :)

File: src/posts/model.js [hunk 4]
```diff
@@ -14,12 +16,10 @@ export const PostBaseData = graphql`
         }
       }
     }
+    id
```
Comment (Moeriki) [inline src/posts/model.js]: Keep id at the top.

File: src/posts/model.js (hunk 2 above)
Comment (naseerkassar) [reply src/posts/model.js]: Fixed :)
Thanks for the feedback

File: src/posts/model.js (hunk 3 above)
Comment (naseerkassar) [reply src/posts/model.js]: Fixed :)

File: src/posts/model.js (hunk 4 above)
Comment (naseerkassar) [reply src/posts/model.js]: Fixed :)

**Pull request merged**